lib/auth/app.ex

```diff
@@ -0,0 +1,147 @@
+defmodule Auth.App do
+  @moduledoc """
+  Schema and helper functions for creating/managing Apps.
+  """
+  use Ecto.Schema
+  import Ecto.Changeset
+  import Ecto.Query, warn: false
+  alias Auth.Repo
+  # https://stackoverflow.com/a/47501059/1148249
+  alias __MODULE__
+
+  schema "apps" do
+    field :desc, :binary
+    field :end, :naive_datetime
+    field :name, :binary
+    field :url, :binary
+    field :person_id, :id
+    field :status, :id
+    has_many :apikeys, Auth.Apikey
+
+    timestamps()
+  end
+
+  @doc false
+  def changeset(app, attrs) do
+    app
+    |> cast(attrs, [:name, :desc, :url, :end, :person_id, :status])
+    |> validate_required([:name, :url])
+  end
+
+  @doc """
+  Returns the list of apps.
+
+  ## Examples
+
+      iex> list_apps()
+      [%App{}, ...]
+
+  """
+  def list_apps do
+    App
+    |> where([a], a.status != 6)
+    |> Repo.all()
+  end
+
+  def list_apps(person_id) do
+    App
+    |> where([a], a.status != 6 and a.person_id == ^person_id)
```

Comment:
Using a magic number here is still confusing, these should be refactored into constants

Reply:
Agreed. This can be improved I was just hard-coding it because (_at least for now_) these are known values.
https://github.com/dwyl/auth/blob/f27ce54fb08a7a7f6db804221cfe21cfad435439/priv/repo/statuses.json#L28-L29